File: sudoku-common/common-web/src/main/java/com/zumin/sudoku/common/web/annotation/ComRestController.java
package com.zumin.sudoku.common.web.annotation;

import io.swagger.annotations.Api;
import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.springframework.core.annotation.AliasFor;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * 整合Controller层的注解
 */
@Api
@Validated
@RequestMapping
@RestController
@Inherited
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE})
public @interface ComRestController {

  @AliasFor(annotation = RequestMapping.class)
  String[] path() default {};

  @AliasFor(annotation = Api.class)
  String[] tags() default {""};
}

File: sudoku-common/common-web/src/main/java/com/zumin/sudoku/common/web/ComRestController.kt
package com.zumin.sudoku.common.web

import io.swagger.annotations.Api
import org.springframework.core.annotation.AliasFor
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.lang.annotation.Inherited

/**
 * 整合Controller层的注解
 */
@Api
@Validated
@RequestMapping
@RestController
@Inherited
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class ComRestController(
  @get:AliasFor(annotation = RequestMapping::class)
  val path: Array<String> = [],
  @get:AliasFor(annotation = Api::class)
  val tags: Array<String> = [""]
)